packages/@lwc/engine-core/src/framework/renderer.ts

```diff
@@ -9,6 +9,7 @@ export type HostNode = any;
 export type HostElement = any;
 
 export interface Renderer<N = HostNode, E = HostElement> {
+    ssr: boolean;
```

Comment:
this smells! why is this needed? in theory, different runtimes will rely on different renderers. Now, depending on the renderer, the consumer of the renderer object will behave differently based on the renderer itself. I know that we already do that for syntheticShadow, but eventually that will go away (once we finish the node reactions and slot polyfills), while `ssr` is not a temporary measurement, it seems.

Reply:
For SSR we will need to disable the 3 following aspects of the engine:
- reactivity 
- `renderedCallback` invocation
- `wire` adapters invocation

While I agree that adding a `ssr` flag on the renderer (the same way we have a `syntheticShadow` flag) is not an ideal nor elegant approach, it allows to toggle those behavior without putting too much logic into the renderer. What about providing a new set of flags to turn on/off the different features listed above instead of having a single `ssr` flag?

On the same note, I don't think the `syntheticShadow` flag should go away. In the case of SSR, we will need to know what environment to target (synthetic or native shadow) to know how to serialize the DOM tree into HTML.

Reply:
IMO, there should be one flag. I dislike the complexity of managing the permutation of flags. 
My long term approach is if reactivity is an artifact of the engine-dom, moving that out of engine-core to engine-dom would be the clean approach to eliminate reactivity for ssr.
I agree that this can be done iteratively in an different PR.

Reply:
I think we can go better!

Reply:
We will, I think we can keep on lifting it up, but let's iterate and not hold this PR hostage for now.